### ISSUE TITLE
Command to set specific sound device

### ISSUE THREAD

**Issue by tresf** [RogueAmoeba/Soundflower-Original#26]
Is there a command line parameter to force a particular sound device?  I have a workstation with a shared HDMI device and when switching between devices, Soundflower resets to an undesired device.

I found an old script on Google Code which monitors for the HDMI device change event in `/var/log/system.log` and from what I can read, it will work, but the script is written very poorly and I'd like to update it.

Unfortunately, I can't for the life of me figure out how the writer of the script was able to programatically switch the soundflower's device to HDMI... 

Here's my code so far (untested until I can prove we can force the device properly):
https://gist.github.com/tresf/48b4abe5c551edeadf2b

Help or a pointer into the source which processes this param is appreciated.  I'm not afraid to fire up XCode too if needed.  Thanks in advance!

-@tresf

**Comment by incandescentman** [comment]
I'm interested in this too. Did you ever figure it out?

**Comment by tresf** [comment]
No, I bought a product called "Sound Control" from Static-Z  due to bugs (audio distorting  and cutting out) like and device prioritization/preference issues with the Soundflower product.  The developer has been very responsive to bugs and features, so I'll continue using it.

**Comment by incandescentman** [comment]
oh wow, that looks great! does it do everything SoundFlower can do or are there drawbacks?

**Comment by tresf** [comment]
Sorry in advance to the maintainer of this project for using it as a forum.

@incandescentman I only use it for HDMI controls and auto-device switching.  It uses some unorthodox ways of overriding the default volume controls and that has forced me to use some workarounds and 3rd party software for the Apple remote, but at the end of the day, it's the only product I've found that does those two things properly.

Static-Z also makes another product with some more advanced features called "Sound Siphon" that I was using previously, so depending on which Soundflower features you use, you may require two separate applications.

**Comment by incandescentman** [comment]
@tresf Thanks for explaining that. That sounds great. I'm downloading it!
